docs/modules/ROOT/pages/user-guide/native-mode.adoc

```diff
@@ -111,3 +111,28 @@ in {@code application.properties} - e.g.
 quarkus.index-dependency.commons-lang3.group-id = org.apache.commons
 quarkus.index-dependency.commons-lang3.artifact-id = commons-lang3
 ----
+
+[[serialization]]
+== Registration for serialization
+
+If serialization support is requested via `quarkus.camel.native.reflection.serialization-enabled`, following classes are registered for serialization:
+
+* `java.lang.Boolean`
+* `java.lang.Byte`
+* `java.lang.Character`
+* `java.lang.Float`
+* `java.lang.Double`
+* `java.lang.Integer`
+* `java.lang.Long`
+* `java.lang.Number`
+* `java.lang.String`
+* `java.math.BigInteger`
+* `java.util.Date`
+* `java.util.HashMap`
+* `java.util.LinkedHashMap`
+* `org.apache.camel.support.DefaultExchangeHolder`
+
+Developers can configure classes for serialization via the serialization configuration file
```

Comment:
We can address this in a follow up issue / PR but we shouldn't we mention `@RegisterForReflection(serialization = true)` as the preferred way of registering user classes for serialization? `-H:SerializationConfigurationResources` is kind of a low level option which should be avoided if possible.

Maybe this has been discussed elsewhere already but I wonder if we should introduce a config property that lets you configure arbitrary classes for serialization like we have for reflection?

Reply:
Oh I understand one previous comment wrongly, I'll reword it with @RegisterForReflection(serialization = true)

Reply:
done